rust/private/rustc.bzl

```diff
@@ -268,6 +268,7 @@ def collect_inputs(
         files,
         toolchain,
         cc_toolchain,
+        crate_type,
```

Comment:
Is `crate_type` not already a part of `crate_info`?
https://github.com/bazelbuild/rules_rust/blob/ce0291cb5ffa5b2ba27820e225d03f00a51ec1d5/rust/private/providers.bzl#L31

Why does this need to explicitly be added?

Reply:
Chatted with @hlopko, and it seems currently indeed `crate_type` is the same as `crate_info.type`, hence redundant. For some historical context, we think that [crate_info.wrapped_crate_type](https://github.com/bazelbuild/rules_rust/blob/e10ab210f34799aa6bffd7a0f11f587967be9587/rust/private/providers.bzl#L32) obsoleted the need for the crate_type arg. I'll update this PR to use crate_info.type and send a follow-up PR to update the rest of the functionality in rules_rust to use `crate_info.type` instead of passing an additional `crate_type` argument, where it makes sense.